PL2/PL2-parser/src/main/java/nl/tudelft/pl2016gr2/parser/controller/FullGfaReader.java

```diff
@@ -3,153 +3,218 @@
 import nl.tudelft.pl2016gr2.model.Node;
 import nl.tudelft.pl2016gr2.model.OriginalGraph;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Scanner;
+import java.util.HashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 /**
  * This class reads a gfa file.
  *
  * @author Cas
- *
  */
 public class FullGfaReader {
 
-  private Scanner sc;
-
-  private ArrayList<String> genoms = new ArrayList<>();
-
+  private static final int SHIFT_BY_BASE_10 = 10;
+  private final ArrayList<String> genomes = new ArrayList<>();
+  private final HashMap<Integer, Node> nodes = new HashMap<>();
   private OriginalGraph originalGraph;
-  public final int numNodes;
 
   /**
    * Creates a reader object and reads the gfa data from the filename.
    *
-   * @param filename  the name of the file to be read.
-   * @param graphsize the size of the graph.
+   * @param fileName the name of the file to read.
    */
-  public FullGfaReader(String filename, int graphsize) {
-    this.numNodes = graphsize;
-    this.sc = new Scanner(FullGfaReader.class.getClassLoader().getResourceAsStream(filename));
-    originalGraph = new OriginalGraph();
-    prepNodes(originalGraph, numNodes);
-    read();
-
-    sc.close();
-    sc = null;
+  public FullGfaReader(String fileName) {
+    read(fileName);
   }
 
   /**
-   * Initialize the nodes of a graph.
+   * Read a GFA file.
    *
-   * @param graph the graph of which to initialize a set amount of node.
-   * @param nodes the amount of nodes to initialize.
+   * @param fileName the name of the GFA file.
    */
-  protected static void prepNodes(OriginalGraph graph, int nodes) {
-    for (int i = 1; i < nodes + 1; i++) {
-      graph.addNode(new Node(i, 1, new ArrayList<>(), 0));
+  private void read(String fileName) {
+    try {
+      BufferedReader br = new BufferedReader(new InputStreamReader(FullGfaReader.class
+          .getClassLoader().getResourceAsStream(fileName)));
+      br.readLine();
+      String line;
+      while ((line = br.readLine()) != null) {
+        char[] chars = line.toCharArray();
+        switch (chars[0]) {
+          case 'L':
+            parseEdge(chars);
+            break;
+          case 'S':
+            parseNode(chars);
+            break;
+          case 'H':
+            parseHeader(chars);
+            break;
+          default:
```

Comment:
Did you put in an empty default case to silence static tools?
In the default case you could log it, (maybe even throw an exception for it) because it is pretty bad if this happens.